src/handlers/TelegramHandler.ts

```diff
@@ -0,0 +1,156 @@
+        scope: this.SCOPE,
+        nonce: this.nonce,
+      },
+      clonedParams
+    );
+    Object.keys(finalJwtParams).forEach((key: string) => {
+      const localKey = key as keyof typeof finalJwtParams;
+      if (finalJwtParams[localKey]) finalUrl.searchParams.append(localKey, finalJwtParams[localKey]);
+    });
+    this.finalURL = finalUrl;
+  }
+
+  objectToAuthDataMap(tgAuthenticationResult: string) {
+    return JSON.parse(atob(tgAuthenticationResult)) as { first_name: string; last_name: string; photo_url: string; username: string; id: number };
+  }
+
+  async getUserInfo(params: LoginWindowResponse): Promise<TorusVerifierResponse> {
+    const { idToken } = params;
+    const userInfo = this.objectToAuthDataMap(idToken);
+    const { photo_url: profileImage = "", first_name = "", last_name = "", id } = userInfo;
+    return {
+      email: "", // Telegram does not provide email
+      name: `${first_name} ${last_name}`,
+      profileImage,
+      verifier: this.params.verifier,
+      verifierId: id.toString(),
+      typeOfLogin: this.params.typeOfLogin,
+    };
+  }
+
+  async handleLoginWindow(params: { locationReplaceOnRedirect?: boolean; popupFeatures?: string }): Promise<LoginWindowResponse> {
+    const verifierWindow = new PopupHandler({ url: this.finalURL, features: params.popupFeatures, timeout: getTimeout(this.params.typeOfLogin) });
+    if (this.params.uxMode === UX_MODE.REDIRECT) {
+      verifierWindow.redirect(params.locationReplaceOnRedirect);
+    } else {
+      const { BroadcastChannel } = await import("@toruslabs/broadcast-channel");
+      return new Promise<LoginWindowResponse>((resolve, reject) => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let bc: any;
+        const handleData = async (ev: string) => {
+          try {
+            const { event, origin, result, ...rest } = (JSON.parse(ev) as PopupResponse) || {};
+            // 1. Parse URL
+            const parsedUrl = new URL(origin);
+            // 2. Get state param
+            const stateParam = parsedUrl.searchParams.get("state");
+
+            if (event && event === "auth_result") {
+              if (!this.params.redirectToOpener && bc) await bc.postMessage({ success: true });
+              // properly resolve the data
+              resolve({
+                accessToken: "",
+                ...rest,
+                idToken: base64url.encode(JSON.stringify(result)) || "",
+                state: atob(stateParam) as unknown as { [key: string]: string },
+              });
+            }
+          } catch (error) {
+            log.error(error);
+            reject(error);
+          }
+        };
+
+        if (!this.params.redirectToOpener) {
+          bc = new BroadcastChannel<{
+            error: string;
+            data: PopupResponse;
+          }>(`redirect_channel_${this.nonce}`, broadcastChannelOptions);
+          bc.addEventListener("message", async (ev: string) => {
```

Comment:
code here is not matching the abstract login handler. is this using a custom service worker?